api/server/router/build/build_routes.go

```diff
@@ -61,9 +61,21 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 		BuildID:        r.FormValue("buildid"),
 	}
 
-	if runtime.GOOS != "windows" && options.SecurityOpt != nil {
-		// SecurityOpt only supports "credentials-spec" on Windows, and not used on other platforms.
-		return nil, invalidParam{errors.New("security options are not supported on " + runtime.GOOS)}
+	if len(options.SecurityOpt) > 0 {
+		// SecurityOpt only supports "credentialspec" on Windows, and not used on other platforms.
+		if runtime.GOOS != "windows" {
+			return nil, invalidParam{errors.New("security options are not supported on " + runtime.GOOS)}
+		}
+		for _, secOpt := range options.SecurityOpt {
+			opt, _, ok := strings.Cut(secOpt, "=")
+			if !ok {
+				return nil, invalidParam{errors.New("invalid security option: no equals sign in supplied value " + secOpt)}
+			}
+			// FIXME(thaJeztah): options should not be case-insensitive
+			if !strings.EqualFold(opt, "credentialspec") {
+				return nil, invalidParam{errors.New("security option not supported: " + opt)}
+			}
```

Comment:
I think it would be a good idea to report `opt == ""` as an invalid security option. `"security option not supported: "` is not exactly the clearest error message to return when someone provides a security option like `"=foobar"`.